Codegen generates javadoc with broken link

The javadoc erases the parameterized types, then, as an example below, the method link is broken

```java
/**
   * Batch execute
   * <p>
   * Unlike the <em>bare</em> Vert.x variant, this method returns a {@link io.smallrye.mutiny.Uni Uni}.
   * Don't forget to <em>subscribe</em> on it to trigger the operation.
   * @param queryFunction query function
   * @param bindBatchValues bind batch values
   * @return the {@link io.smallrye.mutiny.Uni uni} firing the result of the operation when completed, or a failure if the operation failed.
   */
  @CheckReturnValue
  public io.smallrye.mutiny.Uni<io.github.zero88.jooqx.BatchResult> batch(Function<org.jooq.DSLContext, org.jooq.Query> queryFunction, io.github.zero88.jooqx.BindBatchValues bindBatchValues) { 
    return io.smallrye.mutiny.vertx.AsyncResultUni.toUni(handler -> {
        delegate.batch(queryFunction, bindBatchValues, handler);
    });
  }

/**
   * Blocking variant of {@link io.github.zero88.jooqx.mutiny.SQLBatchExecutor#batch(Function<DSLContext,Query>,BindBatchValues)}.
   * <p>
   * This method waits for the completion of the underlying asynchronous operation.
   * If the operation completes successfully, the result is returned, otherwise the failure is thrown (potentially wrapped in a RuntimeException).
   * @param queryFunction query function
   * @param bindBatchValues bind batch values
   * @return the BatchResult instance produced by the operation.
   */
  public io.github.zero88.jooqx.BatchResult batchAndAwait(Function<org.jooq.DSLContext, org.jooq.Query> queryFunction, io.github.zero88.jooqx.BindBatchValues bindBatchValues) { 
    return (io.github.zero88.jooqx.BatchResult) batch(queryFunction, bindBatchValues).await().indefinitely();
  }
```

Warning in javadoc task

```
io/github/zero88/jooqx/mutiny/Jooqx.java:182: warning - Tag @link: reference not found: io.github.zero88.jooqx.mutiny.SQLBatchExecutor#batch(Function<DSLContext,Query>,BindBatchValues)
```

--- a/vertx-mutiny-generator/src/main/java/io/smallrye/mutiny/vertx/codegen/methods/MutinyMethodGenerator.java
+++ b/vertx-mutiny-generator/src/main/java/io/smallrye/mutiny/vertx/codegen/methods/MutinyMethodGenerator.java
@@ -92,10 +92,25 @@ public class MutinyMethodGenerator {
         }
     }
 
-    private String sanitize(String link) {
-        return link
+    public static String sanitize(String link) {
+        String sanitized = link
                 .replace("AndAwait", "")
                 .replace("AndForget", "");
+
+        // Need to remove the generic parameters
+        StringBuilder buffer = new StringBuilder();
+        boolean param = false;
+        for (char c : sanitized.toCharArray()) {
+            if (c == '<'  && ! param) {
+                param = true;
+            } else if (c == '>'  && param) {
+                param = false;
+            } else if (! param  && c != '>') {
+                buffer.append(c);
+            }
+        }
+        return buffer.toString();
+
     }
 
     public void generateMethodDeclaration(MutinyMethodDescriptor descriptor) {

--- a/vertx-mutiny-generator/src/tck/java/org/extra/JavadocTest.java
+++ b/vertx-mutiny-generator/src/tck/java/org/extra/JavadocTest.java
@@ -1,0 +1,37 @@
+package org.extra;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@VertxGen
+public interface JavadocTest {
+
+
+    /**
+     * Do something.
+     * @param list list
+     * @param handler handler
+     */
+    void doSomething(List<String> list, Handler<AsyncResult<Void>> handler);
+
+    /**
+     * Do something.
+     * @param list1 list
+     * @param list2  list
+     * @param handler handler
+     */
+    void doSomething2(List<String> list1, List<String> list2, Handler<AsyncResult<Void>> handler);
+
+
+    /**
+     * Do something.
+     * @param list list
+     */
+    <T> void doSomething3(List<T> list, Handler<AsyncResult<Void>> handler);
+
+}

